error: controlling expression type 'int64_t' (aka 'long') not compatible with any generic association type

Compilation fails:
```
cc -fno-strict-overflow -Wsign-compare -Wunreachable-code -DNDEBUG -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -fPIC -DNPY_TARGET_VERSION=NPY_2_0_API_VERSION -DNPY_NO_DEPRECATED_API=NPY_2_0_API_VERSION -DPy_LIMITED_API=0x030A0000 -I/usr/local/lib/python3.12/site-packages/numpy/_core/include -Icextern/astrometry.net -Iastropy_healpix -Icextern/numpy -I/usr/local/include/python3.12 -c astropy_healpix/_core.c -o build/temp.freebsd-15.1-STABLE-amd64-cpython-312/astropy_healpix/_core.o -O2
astropy_healpix/_core.c:313:44: error: controlling expression type 'int64_t' (aka 'long') not compatible with any generic association type
  313 |             *out = 63 - stdc_leading_zeros(in);
      |                                            ^~
1 error generated.
```
This patch fixes compilation:
```
--- astropy_healpix/_core.c.orig        2026-07-07 20:08:05 UTC
+++ astropy_healpix/_core.c
@@ -310,7 +310,7 @@ static void bit_scan_reverse_loop(
         int64_t  in = *(int64_t *) &args[0][i * steps[0]];
         int    *out =  (int *)     &args[1][i * steps[1]];
         #if __has_include(<stdbit.h>)  
-            *out = 63 - stdc_leading_zeros(in);
+            *out = 63 - stdc_leading_zeros((unsigned long long)in);
         #elif defined(_MSC_VER)
             unsigned long index;
             if (_BitScanReverse64(&index, in))
```


Version: 2.0.0
FreeBSD 15.1

--- a/astropy_healpix/_core.c
+++ b/astropy_healpix/_core.c
@@ -308,20 +308,26 @@ static void bit_scan_reverse_loop(
     for (i = 0; i < n; i ++)
     {
         int64_t   in = *(int64_t *) &args[0][i * steps[0]];
-        int64_t *out =  (int *)     &args[1][i * steps[1]];
-        int result;
-        #if __has_include(<stdbit.h>)
-            result = 63 - stdc_leading_zeros(in);
-        #elif defined(_MSC_VER)
-            unsigned long index;
-            if (_BitScanReverse64(&index, in))
-                result = index;
-            else
-                result = -1;
-        #else
-            result = 63 - __builtin_clzll(in);
-        #endif
-        *out = result;
+        int64_t *out =  (int64_t *) &args[1][i * steps[1]];
+        if (in <= 0)
+        {
+            *out = -1;
+        } else {
+            int result;
+            uint64_t arg = (uint64_t) in;
+            #if __has_include(<stdbit.h>)
+                result = 63 - stdc_leading_zeros(arg);
+            #elif defined(_MSC_VER)
+                unsigned long index;
+                if (_BitScanReverse64(&index, arg))
+                    result = index;
+                else
+                    result = -1;
+            #else
+                result = 63 - __builtin_clzll(arg);
+            #endif
+            *out = result;
+        }
     }
 }
 

--- a/astropy_healpix/tests/test_core.py
+++ b/astropy_healpix/tests/test_core.py
@@ -42,6 +42,13 @@ def test_nside_to_level(level):
     assert nside_to_level(nside) == level
 
 
+@pytest.mark.parametrize(
+    "level", [0, -1, np.iinfo(np.int32).min, np.iinfo(np.int64).min]
+)
+def test_nside_to_level_invalid(level):
+    assert nside_to_level(level) == -1
+
+
 def test_level_ipix_to_uniq():
     assert 11 + 4 * 4**0 == level_ipix_to_uniq(0, 11)
     assert 62540 + 4 * 4**15 == level_ipix_to_uniq(15, 62540)

